XenModel/Actions/Network/ChangeNetworkingAction.cs

```diff
@@ -195,6 +201,13 @@ private void ReconfigureManagement(bool this_host, int hi)
                                                           clearDownManagementIP);
         }
 
+        private void PoolReconfigureManagement(int hi)
+        {
+            System.Diagnostics.Trace.Assert(newManagement != null);

```

Comment:
I agree that we shouldn't hit this code with null `downManagement`. But that's why assertion is suitable here. In general we use assertion to make sure that we don't hit that code with a certain condition. 
And we shouldn't throw an assertion failure if `newManagement` is null because this can happen, according to the comments on the constructor. I think that for `newManagement` we need a null check, but in an `if` statement, not an assertion.